### PR TITLE
Issue #6103: Allow enforcing revisions

### LINKED ISSUE #6103: "Create new revision by default" doesn't enable revisions
## Description of the bug

The Revision settings for a content type are not clear. The options under Revision settings are:

- Show option to create new revisions
Revisions allow content editors to view changes over time and revert changes if needed.
- Create new revision by default

The current functionality is that the "Show option to create new revisions" actually enables or disables revisions for the content type. If "Show option to create new revisions" is off, the "Create new revision by default" is ignored.

## Steps To Reproduce

To reproduce the behavior:
1. Open the configuration page for a content type.
2. Set the Revision settings to uncheck "Show option to create new revisions" and check "Create new revision by default"
3. Edit a node of the content type that was edited.
4. View the revisions of the edited node.

## Actual behavior

No revisions is saved for the edited node.

## Expected behavior

Since the Revision settings are showing "Show option to create new revisions" unchecked and "Create new revision by default" checked, the revision setting shouldn't be shown on the node edit page, but the revision should be saved.

Alternatively, the Revision settings can be reworded to reflect the actual behavior. Maybe: 
- Enable the option to create new revisions
Revisions allow content editors to view changes over time and revert changes if needed.
- Create new revision by default 
And "Create new revision by default" should be hidden if the first option is disabled.

## Additional information

In my use case, I want revisions saved for every node edit without the option to not save a revision. I am able to accomplish this by enabling both check boxes under Revision settings and using the Simplify module to remove the revision setting from the node edit form.

Add any other information that could help, such as:
- Backdrop CMS version: 1.24.2

### PR DESCRIPTION
**Pull request by klonos**
Fixes https://github.com/backdrop/backdrop-issues/issues/6103

### CODE REVIEW

**Comment by klonos** [comment]
cspell failures are to be addressed in https://github.com/backdrop/backdrop/pull/4544 (I'll rebase this PR here if that other one gets merged first)

File: core/modules/node/node.types.inc [hunk 1]
```diff
@@ -408,26 +413,48 @@ function node_type_form($form, &$form_state, $type = NULL) {
...
     '#weight' => 10,
+    '#description' => $revisions_help,
   );
-  $form['revision']['revision']['revision_enabled'] = array(
-    '#type' => 'checkbox',
-    '#title' => t('Show option to create new revisions'),
-    '#default_value' => $type->settings['revision_enabled'],
-    '#parents' => array('revision_enabled'),
-    '#description' => t('Revisions allow content editors to view changes over time and revert changes if needed.'),
+  $form['revision']['revision']['revisions'] = array(
```
Comment (kswan) [inline core/modules/node/node.types.inc:432]: Line 786 has another 'revision_enabled' that should be changed to 'revisions'.

File: core/modules/node/node.install
Comment (kswan) [inline core/modules/node/node.install:1]: I believe line 477 should be changed from 'revision_enabled' to 'revisions'.

File: core/modules/node/node.types.inc (hunk 1 above)
Comment (klonos) [reply core/modules/node/node.types.inc:432]: Good catch @kswan 👍🏼 

Fixed with [`3f8398e` (#4546)](https://github.com/backdrop/backdrop/pull/4546/commits/3f8398e7e6fed31cab9d2fb0f1890fcd48d5d1a4)

**Comment by klonos** [comment]
Tugboat sandbox failed to be built. Kicking things...

**Comment by klonos** [comment]
...nothing that a good-old rebase can't fix 😉

**Comment by klonos** [comment]
Unrelated random failure - only on php8:
```
Exception Warning    file.inc          2305                                                                            
    is_dir(): Unable to find the wrapper "public" - did you forget to enable it when you configured PHP?
```

**Comment by kswan** [comment]
@klonos Are you able to rebase this easily? I think the cspell errors should be resolved and the tugboat has expired.

I believe I confirmed this PR works and you fixed the items I found in my code review, but I would like to give it another check before updating the issue.

**Comment by herbdool** [comment]
Replaced by https://github.com/backdrop/backdrop/pull/4649 on Issue 6103. But would recommend putting this PR on a new issue where there might be backwards-incompatible changes.

**Pull request closed without merging**